lib/cloudant.js

```diff
@@ -75,7 +75,42 @@ Cloudant.prototype.getTypes = function() {
  */
 Cloudant.prototype.connect = function(cb) {
   debug('Cloudant.prototype.connect');
-  cb(null, this.cloudant);
+  var self = this;
+  // check if cloudant server is up
+  if (self.options.url) {
+    request.get(self.options.url, function(err, res, body) {
+      if (err) cb(res.errMsg);
+    });
+  } else {
+    var opts = {
+      auth: {
+        user: self.options.username,
+        pass: self.options.password,
+        sendImmediately: true,
+      },
+      uri: self.options.username + '.cloudant.com',
+      port: self.options.port || '',
+    };
+    request.get(opts, function(err, res, body) {
+      if (err) cb(res.errMsg);
+    });
+  }
+  // strip db name if defined in path of url before
+  // sending it to our driver
+  if (self.options.url) {
```

Comment:
Looks like this block(L#100-L#113) and the previous one(L#80-L#97) will execute in parallel. 
Maybe wrap this block into a function, e.g. 'function getDriverInstance()', and call it after L#82 and L#95?